# Python `journey_data` insights

In [18]:
# Import libraries
import pandas as pd

# Import data from data folder
trips_data = pd.read_csv("../data/trips_data.csv")
journey_data = pd.read_csv("../data/journey_data.csv")
same_start_end_journey_data = pd.read_csv("../data/same_start_end_journey_data.csv")
diff_start_end_journey_data = pd.read_csv("../data/diff_start_end_journey_data.csv")

Start by inspecting the overall statistics of `trips_data` before comparing with the data of various journey types

In [19]:
trips_data.describe()

Unnamed: 0         bikeid  duration_minutes          month  \
count  622438.000000  622438.000000     622438.000000  622438.000000   
mean   312599.019851     472.333821         28.812683       5.904710   
std    180478.946123     324.315969         77.184854       3.143848   
min         0.000000       3.000000          0.000000       1.000000   
25%    156291.250000     209.000000          8.000000       3.000000   
50%    312600.500000     420.000000         14.000000       6.000000   
75%    468888.750000     745.000000         27.000000       9.000000   
max    625200.000000    5089.000000      10981.000000      12.000000   

            trip_id           year     start_hour  start_station_id  \
count  6.224380e+05  622438.000000  622438.000000     622438.000000   
mean   5.226115e+09    2015.453052      14.296674       2581.991199   
std    4.936609e+09       1.050531       4.517494        326.642565   
min    8.269930e+06    2013.000000       0.000000       1001.000000   
25%    1.257626e+07    2015.000000      12.000000       2501.000000   
50%    9.900018e+09    2015.000000      15.000000       2549.000000   
75%    9.900186e+09    2016.000000      17.000000       2571.000000   
max    9.900353e+09    2017.000000      23.000000       3687.000000   

       end_station_id            day  
count   622438.000000  622438.000000  
mean      2579.214492       3.424879  
std        327.661555       2.006175  
min       1001.000000       0.000000  
25%       2499.000000       2.000000  
50%       2548.000000       4.000000  
75%       2571.000000       5.000000  
max       3687.000000       6.000000

In [20]:
journey_data[:15]

start_station_name  \
0                       Riverside @ S. Lamar   
1                       Rainey St @ Cummings   
2                             2nd & Congress   
3                   City Hall / Lavaca & 2nd   
4          Capitol Station / Congress & 11th   
5                                Zilker Park   
6              Pfluger Bridge @ W 2nd Street   
7                 Barton Springs & Riverside   
8                          Palmer Auditorium   
9                     Davis at Rainey Street   
10                       Barton Springs Pool   
11       Long Center @ South 1st & Riverside   
12   Convention Center / 4th St. @ MetroRail   
13  MoPac Pedestrian Bridge @ Veterans Drive   
14                               5th & Bowie   

                            end_station_name  day     parts_of_day  \
0                       Riverside @ S. Lamar    6  early afternoon   
1                       Rainey St @ Cummings    5  early afternoon   
2                             2nd & Congress    5  early afternoon   
3                   City Hall / Lavaca & 2nd    5  early afternoon   
4          Capitol Station / Congress & 11th    6   late afternoon   
5                                Zilker Park    5   late afternoon   
6              Pfluger Bridge @ W 2nd Street    6   late afternoon   
7                 Barton Springs & Riverside    5  early afternoon   
8                          Palmer Auditorium    6  early afternoon   
9                     Davis at Rainey Street    5   late afternoon   
10                       Barton Springs Pool    6  early afternoon   
11       Long Center @ South 1st & Riverside    5  early afternoon   
12   Convention Center / 4th St. @ MetroRail    5  early afternoon   
13  MoPac Pedestrian Bridge @ Veterans Drive    5  early afternoon   
14                            4th & Congress    3    early morning   

    duration_minutes  month subscriber_type         year  start_hour  count  
0          49.109894      7         Walk Up  2015.456959   14.908322   8399  
1          55.968623      7         Walk Up  2015.611493   14.173453   5673  
2          64.688035      5         Walk Up  2015.417033   13.461467   4321  
3          58.751687      3         Walk Up  2015.151501   13.902723   4297  
4          44.236900      7         Walk Up  2015.097077   14.570601   3626  
5          53.530878      3         Walk Up  2016.215730   14.532539   3611  
6          51.269375      7         Walk Up  2015.562086   14.486891   3471  
7          65.467645      7         Walk Up  2015.071047   14.663442   2998  
8          56.798149      3         Walk Up  2015.394106   14.227553   2918  
9          52.519453      3         Walk Up  2015.572029   14.150719   2853  
10         47.563316      3         Walk Up  2015.399647   14.208466   2835  
11         57.213749      3         Walk Up  2015.511636   14.786251   2793  
12         56.826368      3         Walk Up  2015.120821   13.940729   2632  
13         58.277461      6         Walk Up  2016.369061   14.100342   2631  
14          8.947801      1        Local365  2015.309083   11.785450   2433

Since the top 14 journey have the same start and ending station, it may be more insightful to inspect the datasets with the same start and ending station and different start and ending stations respectively.

In [21]:
same_start_end_journey_data[:15]

start_station_name  \
0                       Riverside @ S. Lamar   
1                       Rainey St @ Cummings   
2                             2nd & Congress   
3                   City Hall / Lavaca & 2nd   
4          Capitol Station / Congress & 11th   
5                                Zilker Park   
6              Pfluger Bridge @ W 2nd Street   
7                 Barton Springs & Riverside   
8                          Palmer Auditorium   
9                     Davis at Rainey Street   
10                       Barton Springs Pool   
11       Long Center @ South 1st & Riverside   
12   Convention Center / 4th St. @ MetroRail   
13  MoPac Pedestrian Bridge @ Veterans Drive   
14                            4th & Congress   

                            end_station_name  day     parts_of_day  \
0                       Riverside @ S. Lamar    6  early afternoon   
1                       Rainey St @ Cummings    5  early afternoon   
2                             2nd & Congress    5  early afternoon   
3                   City Hall / Lavaca & 2nd    5  early afternoon   
4          Capitol Station / Congress & 11th    6   late afternoon   
5                                Zilker Park    5   late afternoon   
6              Pfluger Bridge @ W 2nd Street    6   late afternoon   
7                 Barton Springs & Riverside    5  early afternoon   
8                          Palmer Auditorium    6  early afternoon   
9                     Davis at Rainey Street    5   late afternoon   
10                       Barton Springs Pool    6  early afternoon   
11       Long Center @ South 1st & Riverside    5  early afternoon   
12   Convention Center / 4th St. @ MetroRail    5  early afternoon   
13  MoPac Pedestrian Bridge @ Veterans Drive    5  early afternoon   
14                            4th & Congress    5  early afternoon   

    duration_minutes  month subscriber_type         year  start_hour  count  
0          49.109894      7         Walk Up  2015.456959   14.908322   8399  
1          55.968623      7         Walk Up  2015.611493   14.173453   5673  
2          64.688035      5         Walk Up  2015.417033   13.461467   4321  
3          58.751687      3         Walk Up  2015.151501   13.902723   4297  
4          44.236900      7         Walk Up  2015.097077   14.570601   3626  
5          53.530878      3         Walk Up  2016.215730   14.532539   3611  
6          51.269375      7         Walk Up  2015.562086   14.486891   3471  
7          65.467645      7         Walk Up  2015.071047   14.663442   2998  
8          56.798149      3         Walk Up  2015.394106   14.227553   2918  
9          52.519453      3         Walk Up  2015.572029   14.150719   2853  
10         47.563316      3         Walk Up  2015.399647   14.208466   2835  
11         57.213749      3         Walk Up  2015.511636   14.786251   2793  
12         56.826368      3         Walk Up  2015.120821   13.940729   2632  
13         58.277461      6         Walk Up  2016.369061   14.100342   2631  
14         51.477166      3         Walk Up  2015.196329   14.359795   2343

In [22]:
same_start_end_journey_data[:15].describe()

day  duration_minutes      month         year  start_hour  \
count  15.000000         15.000000  15.000000    15.000000   15.000000   
mean    5.333333         54.913247   4.666667  2015.476437   14.298220   
std     0.487950          5.861636   1.914854     0.379344    0.376656   
min     5.000000         44.236900   3.000000  2015.071047   13.461467   
25%     5.000000         51.373270   3.000000  2015.173915   14.125530   
50%     5.000000         55.968623   3.000000  2015.417033   14.227553   
75%     6.000000         57.745605   7.000000  2015.567058   14.551570   
max     6.000000         65.467645   7.000000  2016.369061   14.908322   

             count  
count    15.000000  
mean   3693.400000  
std    1565.202169  
min    2343.000000  
25%    2814.000000  
50%    2998.000000  
75%    3961.500000  
max    8399.000000

From the top 15 `same_start_end_journey_data`,
1. top trips mostly take place on the Weekends 
2. The average duration is relatively high (range of 44.2 to 65.5 minutes) with mean 54.9 minutes compared to the mean of 28.8 minutes for all trips
3. Most common subscriber type is Walk Up
4. The average starting hour is during the afternoon at 2:18pm (14.29hrs)

From these information, we can infer that **casual cyclers are a large proportion of Bikeshare's market**. These users choose to cycle for long periods during the weekend--potentially to visit places around Austin before returning the bike at the same starting point. **A potential market of Bikeshare is therefore tourists**

In [23]:
diff_start_end_journey_data[:15]

start_station_name  \
0                               5th & Bowie   
1                            4th & Congress   
2                               5th & Bowie   
3                                3rd & West   
4   Convention Center / 4th St. @ MetroRail   
5                  South Congress & Academy   
6                  City Hall / Lavaca & 2nd   
7     Republic Square @ Guadalupe & 4th St.   
8                      Rainey St @ Cummings   
9                               5th & Bowie   
10                               3rd & West   
11                     Riverside @ S. Lamar   
12                 City Hall / Lavaca & 2nd   
13                 City Hall / Lavaca & 2nd   
14                     Riverside @ S. Lamar   

                           end_station_name  day     parts_of_day  \
0                            4th & Congress    3    early morning   
1                               5th & Bowie    1   late afternoon   
2                  City Hall / Lavaca & 2nd    0   late afternoon   
3                  City Hall / Lavaca & 2nd    4     late morning   
4                  City Hall / Lavaca & 2nd    2    early morning   
5                            2nd & Congress    5   late afternoon   
6                               5th & Bowie    3   late afternoon   
7                               5th & Bowie    2     late morning   
8                  City Hall / Lavaca & 2nd    4     late morning   
9   Convention Center / 4th St. @ MetroRail    4   late afternoon   
10                           4th & Congress    4     late morning   
11                 City Hall / Lavaca & 2nd    5   late afternoon   
12                     Riverside @ S. Lamar    5   late afternoon   
13  Convention Center / 4th St. @ MetroRail    0   late afternoon   
14                     Rainey St @ Cummings    5  early afternoon   

    duration_minutes  month                     subscriber_type         year  \
0           8.947801      1                            Local365  2015.309083   
1          11.289637      3                            Local365  2015.360939   
2          11.005179      3                            Local365  2015.175612   
3           8.807835      3                            Local365  2015.651668   
4          13.819351      3                            Local365  2015.331096   
5          17.380274      3                             Walk Up  2015.395097   
6          10.989833      3                            Local365  2015.102273   
7           7.727437      3                            Local365  2014.670878   
8          17.390152     10                            Local365  2015.536616   
9          13.631126      3  Annual Membership (Austin B-cycle)  2014.930464   
10          9.151333      3                            Local365  2015.672667   
11         19.034459      3                             Walk Up  2015.383108   
12         18.595774      3                             Walk Up  2015.420586   
13         18.637744      3                            Local365  2015.156905   
14         31.316854      6                             Walk Up  2015.674157   

    start_hour  count  
0    11.785450   2433  
1    15.014172   2258  
2    14.809322   2124  
3    13.061408   1889  
4    11.307047   1788  
5    13.399088   1754  
6    14.084928   1672  
7    11.279182   1662  
8    12.492424   1584  
9    14.737086   1510  
10   12.350000   1500  
11   14.039865   1480  
12   14.565099   1467  
13   15.342010   1383  
14   13.958801   1335

In [24]:
diff_start_end_journey_data[:15].describe()

day  duration_minutes      month         year  start_hour  \
count  15.000000         15.000000  15.000000    15.000000   15.000000   
mean    3.133333         14.514986   3.533333  2015.318077   13.481725   
std     1.767430          6.139294   2.030717     0.279832    1.366041   
min     0.000000          7.727437   1.000000  2014.670878   11.279182   
25%     2.000000         10.070583   3.000000  2015.166259   12.421212   
50%     4.000000         13.631126   3.000000  2015.360939   13.958801   
75%     4.500000         17.992963   3.000000  2015.478601   14.651092   
max     5.000000         31.316854  10.000000  2015.674157   15.342010   

             count  
count    15.000000  
mean   1722.600000  
std     326.817511  
min    1335.000000  
25%    1490.000000  
50%    1662.000000  
75%    1838.500000  
max    2433.000000

From the top 15 `diff_start_end_journey_data`,
1. top trips mostly take place on Weekdays (mean is 3.13 which is appoximately Thursday) 
2. The average duration is relatively low (range of 7.7 to 31.3 minutes) with mean 14.5 minutes compared to the mean of 28.8 minutes
4. Most common subscriber type is Local365, followed by Walk Up
5. The average starting hour is around 1:28pm (13.48hrs) but with a larger distribution compared to `journey_data` (1.366 compared to 0.377)

From these information, we can infer that another market of Bikeshare are Local365 i.e. **yearly subscribers according to their bike pass website: https://austin.bcycle.com/passes, who may use the bike to travel short distances frequently**. A potential scenario would be going to work, which makes sense since City Hall / Lavaca & 2nd is a civic district with many offices. **A potential market of Bikeshare is therefore local office workers.**